sympy/functions/special/delta_functions.py

```diff
@@ -99,16 +101,21 @@ def simplify(self, x):
         """
         from sympy.polys.polyroots import roots
 
-        if not self.args[0].has(x) or (len(self.args) > 1 and self.args[1] != 0 ):
+        if 'wrt' not in hints:
+          raise TypeError("The 'wrt' keyword is required for the DiracDelta hint.")
```

Comment:
@smichr This is also a good idea. The error message is great and descriptive. We can add this.